### PR TITLE
MODPUBSUB-87 - PubSub client: create method for module unregistering

### PR DESCRIPTION
**Pull request by RuslanLavrov**

### CODE REVIEW

File: mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java [hunk 1]
```diff
@@ -169,6 +174,47 @@ private PubSubClientUtils() {
     return publishersResult;
   }
 
+  /**
+   * Util method to unregister external module in PubSub.
+   *
+   * @param params - okapi connection params
+   * @return future with true if module was unregistered successfully
+   */
+  public static CompletableFuture<Boolean> unregisterModule(OkapiConnectionParams params) {
```
Comment (KaterynaSenchenko) [inline mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java:183]: @RuslanLavrov, do you think it is worth having separate util methods for unregistering as subscriber and as publisher?

File: mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java [hunk 2]
```diff
@@ -169,6 +174,47 @@ private PubSubClientUtils() {
...
+    Future.succeededFuture()
+      .compose(ar -> unregisterModuleByIdAndRole(client, moduleId, PUBLISHER))
+      .compose(ar -> unregisterModuleByIdAndRole(client, moduleId, SUBSCRIBER))
+      .onFailure(future::completeExceptionally)
+      .onSuccess(ar -> future.complete(true))
+      .onComplete(ar -> client.close());
+    return future;
+  }
+
+  private static Future<Boolean> unregisterModuleByIdAndRole(PubsubClient client, String moduleId, MessagingModule.ModuleRole moduleRole) {
```
Comment (KaterynaSenchenko) [inline mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java:197]: Probably better to use CompletableFuture here as well

File: mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java [hunk 3]
```diff
@@ -169,6 +174,47 @@ private PubSubClientUtils() {
...
+    Promise<Boolean> promise = Promise.promise();
+    try {
+      LOGGER.info("Trying to unregister module with name '{}' as {}", moduleId, moduleRole);
+      client.deletePubsubMessagingModules(moduleId, moduleRole.value(), response -> {
+        if (response.statusCode() == HttpStatus.HTTP_NO_CONTENT.toInt()) {
+          LOGGER.info("Module was successfully unregistered as '{}'", moduleRole);
```
Comment (VRohach) [inline mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java:203]: I think it would be better to add here moduleId as well.

**Review by VRohach — APPROVED** [review]

File: mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java (hunk 1 above)
Comment (RuslanLavrov) [reply mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java:183]: @KaterynaSenchenko, according to the description, there is necessary to unregister a module in pubsub when the module is disabled for a tenant. So, as I understand all information about module publications and subscriptions should be deleted after module disabling. @KaterynaSenchenko, @OleksiiKuzminov, @VRohach, let me know please if you think that it is worth to add them. I will do it.

**Review by KaterynaSenchenko — APPROVED** [review]

**Pull request merged**